storageservice/modules/files/file_client.bal

```diff
@@ -384,6 +430,49 @@ public isolated client class FileClient {
         }
     }
 
+
+    # Downloads a file from fileshare to a specified location as an Byte array
+    #
+    # + fileShareName - Name of the FileShare
+    # + fileName - Name of the file
+    # + azureDirectoryPath - Path of azure directory
+    # + return -  If successful, file content as a byte array. Else an error
+    @display {label: "Download File"}
+    remote isolated function getFileAsByteArray(@display {label: "File Share Name"} string fileShareName, 
+                                     @display {label: "Azure File Name"} string fileName,  
+                                     @display {label: "Azure Directory Path"} string? azureDirectoryPath = ()) 
+                                     returns @display {label: "Response"} byte[]|error {
+        string requestPath = azureDirectoryPath is () ? (SLASH + fileShareName + SLASH + fileName) : (SLASH 
+            + fileShareName + SLASH + azureDirectoryPath + SLASH + fileName);    
+        http:Request request = new;
+        if (self.azureConfig.authorizationMethod === ACCESS_KEY) {
+            map<string> requiredURIParameters ={}; 
+            string resourcePathForSharedkeyAuth = azureDirectoryPath is () ? (fileShareName + SLASH + fileName) 
+                : (fileShareName + SLASH + azureDirectoryPath + SLASH + fileName);
+            AuthorizationDetail  authorizationDetail = {
+                azureRequest: request,
+                azureConfig: self.azureConfig,
+                httpVerb: http:HTTP_GET,
+                resourcePath: resourcePathForSharedkeyAuth,
+                requiredURIParameters: requiredURIParameters
+            };
+            check prepareAuthorizationHeaders(authorizationDetail);      
+        } else {
+            requestPath = requestPath.concat(QUESTION_MARK, self.azureConfig.accessKeyOrSAS.substring(1)); 
+        }
+        map<string> headerMap = populateHeaderMapFromRequest(request);
+        http:Response response = <http:Response> check self.httpClient->get(requestPath, headerMap);
+        if (response.statusCode === http:STATUS_OK ) {
```

Comment:
Do we need this check? If a response other than `http:STATUS_OK` is returned, it is interpreted as an error and will be handled by `check`. Is my understanding wrong there?